### PR TITLE
Pyapp

### PR DESCRIPTION
**Pull request by vishalkor**
Hi Sunil,
In this branch i have added python appium script along with .gitignore and test_requirement.txt file.
Please review the same.

### CODE REVIEW

File: readme [hunk 1]
```diff
@@ -0,0 +1,41 @@
+# Appium
```
Comment (ersuniljoshi) [inline readme]: Change the file name to "ReadMe.md" and follow the proper syntax of ReadMe.md

File: library/element_codexes.py [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+            return driver.find_element_by_name(codex_details[1])
+        elif codex_details[0] == "class":
+            return driver.find_element_by_class_name(codex_details[1])
+        elif codex_details[0] == "xpath":
+            return driver.find_element_by_xpath(codex_details[1])
+        else:
+            return False
+
+
+    def isElementLocated(self,codex_details):
```
Comment (ersuniljoshi) [inline library/element_codexes.py]: why codex_details is a list, we are not using any other apart from codex_details[1]

File: conftest.py [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+    desired_caps = {}
+    desired_caps['platformName'] = 'Android'
+    desired_caps['platformVersion'] = '5.1.1'
+    desired_caps['deviceName'] = device
+    desired_caps['appPackage'] = 'com.surveymonkey'
+    desired_caps['appActivity'] = 'com.surveymonkey.login.activities.LandingActivity'
+    driver = webdriver.Remote('http://localhost:4444/wd/hub', desired_caps)
+    return driver
+
+# @pytest.fixture(scope="function")
```
Comment (ersuniljoshi) [inline conftest.py]: Remove this code, please it has nothing to do with Appium

File: conftest.py [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+def pytest_addoption(parser):
+    parser.addoption("--device", action="store", default="Android Emulator",
+        help="my option: Android Emulator or Android device")
+    parser.addoption("--codexFile", action="store", default="android",
+        help="my option: ios or android or web")
+    parser.addoption("--browser", action="store", default="ff",
+        help="my option: ff or chrome")
+
+
+@pytest.fixture
```
Comment (ersuniljoshi) [inline conftest.py]: What are the default values for this fixture, if we forgot to pass them??

File: Results/assets/style.css [hunk 5]
```diff
@@ -0,0 +1,157 @@
+body {
```
Comment (ersuniljoshi) [inline Results/assets/style.css]: Please remove, all the images , css and html file from the repo. You can create a folder for sample reports and add data to it.

File: test_requirement.txt [hunk 6]
```diff
@@ -0,0 +1,20 @@
+ansi2html==1.1.1
```
Comment (ersuniljoshi) [inline test_requirement.txt]: Change the name of the file to "test-requirements.txt"

File: readme [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+    default : "android"
+    
+Pytest built-in Command Line Parameters/Options for project :
+=============================================================
+
+--html
+    usage  : Generate html report for test result
+    option : Path-for-report-file
+             In our case report will be generate in this path
+             "/home/vishal/gitrepo/appiumDemo/Results/report.html"
```
Comment (ersuniljoshi) [inline readme]: No hardcoded examples please.

File: library/element_codexes.py [hunk 8]
```diff
@@ -0,0 +1,110 @@
...
+
+    def line_is_blank(self, line):
+        if self.whitespace_pattern.match(line) or \
+           self.newline_only_pattern.match(line):
+            return True
+        else:
+            return False
+
+    def read_codex(self, input_file):
+        codex_file_path = os.path.abspath(os.path.join("/home/vishal/forAppiumDemo/demo/newAndroidWebDemo/resources/",input_file))
```
Comment (ersuniljoshi) [inline library/element_codexes.py]: No hardcoding.

File: conftest.py [hunk 9]
```diff
@@ -0,0 +1,73 @@
...
+def device(request):
+    return request.config.getoption("--device")
+
+@pytest.fixture
+def codexFile(request):
+    return request.config.getoption("--codexFile")
+
+@pytest.fixture
+def browser(request):
```
Comment (ersuniljoshi) [inline conftest.py]: Do we need browser fixture for appium?

File: conftest.py [hunk 10]
```diff
@@ -0,0 +1,73 @@
+# Using appium webdriver to automate
+from appium import webdriver
+import selenium
+import pytest
+import os
+
+def pytest_addoption(parser):
+    parser.addoption("--device", action="store", default="Android Emulator",
+        help="my option: Android Emulator or Android device")
```
Comment (ersuniljoshi) [inline conftest.py]: We donot have --device as iOS, but we have codex file as iOS.. :)

File: conftest.py [hunk 11]
```diff
@@ -0,0 +1,73 @@
...
+# @pytest.fixture(scope="function")
+# def setUpWeb(browser):
+#     if browser == 'ff':
+#         driver = selenium.webdriver.Firefox()
+#     else:
+#         driver = selenium.webdriver.Chrome("/home/vishal/Downloads/chromedriver")
+#     return driver
+
+@pytest.fixture(scope="function")
+def setUpWeb(browser):
```
Comment (ersuniljoshi) [inline conftest.py]: Do we need this in Appium?

File: library/element_codexes.py (hunk 2 above)
Comment (vishalkor) [inline library/element_codexes.py]: Here we are using both codex_details[0] and codex_details[1], So it should be list.

File: conftest.py (hunk 11 above)
Comment (vishalkor) [inline conftest.py]: Removed.

File: conftest.py (hunk 9 above)
Comment (vishalkor) [inline conftest.py]: Removed.

File: test_requirement.txt (hunk 6 above)
Comment (vishalkor) [inline test_requirement.txt]: Changed.

File: conftest.py (hunk 4 above)
Comment (vishalkor) [inline conftest.py]: I have removed this fixture because we don't need that in Appium.

File: conftest.py (hunk 3 above)
Comment (vishalkor) [inline conftest.py]: Removed.

File: Results/assets/style.css (hunk 5 above)
Comment (vishalkor) [inline Results/assets/style.css]: I have deleted Results/ directory and added sample reports to sample_reports directory.

File: conftest.py (hunk 10 above)
Comment (vishalkor) [inline conftest.py]: There is no iOS codex file.

File: library/element_codexes.py (hunk 8 above)
Comment (vishalkor) [inline library/element_codexes.py]: I have removed the hard coded path of codex file.

File: readme (hunk 7 above)
Comment (vishalkor) [inline readme]: Removed.

File: readme (hunk 1 above)
Comment (vishalkor) [inline readme]: I have changed readme file to README.md and followed all the markdown syntax.

**Pull request closed without merging**